app/src/main/java/chat/rocket/android/chatdetails/presentation/ChatDetailsPresenter.kt

```diff
@@ -90,6 +92,25 @@ class ChatDetailsPresenter @Inject constructor(
         navigator.toFavoriteMessageList(chatRoomId)
     }
 
+    fun leave(chatRoomId: String, chatRoomType: String) {
+        launchUI(strategy) {
+            try {
+                retryIO("leaveChat($chatRoomId, $chatRoomType") {
+                    client.leaveChat(chatRoomId, roomTypeOf(chatRoomType))
+                }
+                view.showMessage(R.string.left_chat)
+                navigator.toChatRooms()
```

Comment:
At this point when I'm sent back to the chatRooms View, the private group that I just left is still in the recyclerView, and clicking it results in an empty room, and the option to leave it again, etc....  I think you will need to refresh the view after a user leaves a room.

Reply:
@ear-dev This is separate issue. Whenever chatroom is added or removed it gets updated in realtime using subscriptions, but it is not working. I think this issue is also in upstream.

Reply:
Yes @ear-dev I will try to fix this in this PR only. Then we can merge.

Reply:
@ear-dev I was debugging this issue. and got to know that. when room is removed, in the subscription server sends update with type `updated` instead of `removed`. That's why it is not being deleted 

`[StreamMessage(type=chat.rocket.core.internal.realtime.socket.model.Type$Updated@5ea5fa7, data=Room(id=Zirt5gb2bsBBv9KNZ, type=p, user=SimpleUser(id=kMWuK9HCXi9gFfamT, username=Shailesh351, name=null), name=testttiiing1, fullName=testttiiing1, readonly=false, updatedAt=1567402532496, topic=null, description=null, announcement=null, lastMessage=Message(id=RNQ9DXzAhqdEz5mPN, roomId=Zirt5gb2bsBBv9KNZ, message=hi, timestamp=1567402492243, sender=SimpleUser(id=kMWuK9HCXi9gFfamT, username=Shailesh351, name=Shailesh351), updatedAt=1567402492280, editedAt=null, editedBy=null, senderAlias=null, avatar=null, type=null, groupable=false, parseUrls=false, urls=null, mentions=[], channels=[], attachments=null, pinned=false, starred=null, reactions=null, role=null, synced=true, unread=true), broadcast=false, muted=null))]`

https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/blob/b4aa0a4100fadfad25be4e6ac49bbbe0366ecfcf/core/src/main/kotlin/chat/rocket/core/internal/realtime/socket/message/collection/StreamNotifyUser.kt#L77

Reply:
Created upstream issue https://github.com/RocketChat/Rocket.Chat.Android/issues/2479